extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/RuleEliminateMarker.scala

```diff
@@ -17,11 +17,57 @@
 
 package org.apache.kyuubi.plugin.spark.authz.util
 
-import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import java.io.IOException
+
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.catalog.HiveTableRelation
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, Statistics}
 import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.command.DDLUtils
+import org.apache.spark.sql.internal.SQLConf
 
-class RuleEliminateMarker extends Rule[LogicalPlan] {
+class RuleEliminateMarker(spark: SparkSession) extends Rule[LogicalPlan] {
   override def apply(plan: LogicalPlan): LogicalPlan = {
-    plan.transformUp { case rf: RowFilterAndDataMaskingMarker => rf.table }
+    plan.transformUp {
+      case rf: RowFilterAndDataMaskingMarker =>
+        val table = rf.table
+        table match {
+          case relation: HiveTableRelation =>
+            if (DDLUtils.isHiveTable(relation.tableMeta) && relation.tableMeta.stats.isEmpty) {
+              hiveTableWithStats(relation)
+            } else {
+              relation
+            }
+          case _ =>
+            table
+        }
+    }
+  }
+
+  private def hiveTableWithStats(relation: HiveTableRelation): HiveTableRelation = {
```

Comment:
If we do this, table stats should be wrong?

Reply:
The  logical come from DetermineTableStats, it is executed in analyzer, because we warp hiveTableRelation, so it can't be executed.